File: Hackathon.Common/Models/GetFilterModel.cs
using Hackathon.Common.Models.Base;

namespace Hackathon.Common.Models
{
    public class GetFilterModel<T> where T: IFilterModel
    {
        public T Filter { get; set; }

        public int? Page { get; set; }
        public int? PageSize { get; set; }

        public string SortBy { get; set; }
        public SortOrder SortOrder { get; set; } = SortOrder.Asc;
    }
}

File: Hackathon.UI/src/app/models/GetFilterModel.ts
export class GetFilterModel<T>{
  Filter!: T;
  Page:number = 0;
  PageSize: number = 300;
  SortBy!: string;
  SortOrder: SortOrder = SortOrder.Asc;
}

export enum SortOrder {
  Asc = 0,
  Desc = 1
}

